source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -100,6 +100,10 @@ class AuthenticatorImpl : public Logger::Loggable<Logger::Id::jwt>,
   const bool is_allow_failed_;
   const bool is_allow_missing_;
   TimeSource& time_source_;
+
+  // allow 5 seconds of slack when determining token expery
+  const uint64_t jwt_exp_slack = 5;
```

Comment:
Please follow envoy style:
for constant,  define it at the head of cc file, inside annoymous name space, similar to

constexpr uint64_t TimeoutPrecisionFactor = 100;

Reply:
How do I use filter config?

Reply:
1) add new fields in [Provider](https://github.com/envoyproxy/envoy/blob/master/api/envoy/extensions/filters/http/jwt_authn/v3/config.proto#L55) 

2) you can get provider() as

jwks_data_->getJwtProvider()

as in [line](https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/jwt_authn/authenticator.cc#L209)

Reply:
Working on it. Think I need a proper build environment. Is there some documentation to set it up? 
Can't find it in https://github.com/envoyproxy/envoy/

Reply:
You can check out this developer [doc](https://github.com/envoyproxy/envoy/blob/master/DEVELOPER.md)

The easiest way is to use [docker](https://github.com/envoyproxy/envoy/tree/master/ci)

Reply:
Ah. Good. Will do the docker thingy.